astropy/utils/tests/test_collections.py

```diff
@@ -31,3 +33,35 @@ def test_homogeneous_list4():
 def test_homogeneous_list5():
     l = collections.HomogeneousList(int, [1, 2, 3])
     l[1] = 5.0
+
+
+def test_homogeneous_list_setitem_works():
+    l = collections.HomogeneousList(int, [1, 2, 3])
+    l[1] = 5
+
+
+def test_homogeneous_list_setitem_works_with_slice():
+    l = collections.HomogeneousList(int, [1, 2, 3])
+    l[0:1] = [10, 20, 30]
```

Comment:
Shouldn't the slice be `[:]` or `[0:3]`? Why is it set to `[0:1]`? (You can tell that I never used it, LoL)

Reply:
The `[0:1]` replaces the first element with the sequence, `[0:3]` would replace the first, second and third element with the sequence and `[:]` would replace the whole sequence.

In this case the result is `[10, 20, 30, 2, 3]`, for `[0:3]` it would be `[10, 20, 30]` and for `[:]` it would also be `[10, 20, 30]`. I have no preference which one should be tested, I just picked one (the one which I though of first) because all defers to `list`s method and it's just important to check that it gets there. :)

Reply:
Hmm, okay, that is fine.  I was wondering why none of the tests check for the resultant values.

Reply:
oh, I can add result checks.

Reply:
Why not? It doesn't take much more time to do the checks and makes the intent clear. Thanks!

Reply:
better now? :)

Reply:
Yes, thanks! :smile: